index.js

```diff
@@ -1,14 +1,23 @@
 'use strict';
+
 require('harmony-reflect');
 
-module.exports = function (obj) {
-	return new Proxy(obj, {
-		get: function (target, name, receiver) {
+module.exports = function (target) {
+	var handler = {
+		get: function (target, name) {
 			if (!(name in target)) {
 				throw new TypeError('Unknown property: ' + name);
 			}
 
-			return Reflect.get(target, name, receiver);
+			return Reflect.get(target, name);
 		}
-	});
+	};
+
+	var instance = new Proxy(target, handler);
+
+	instance.decorateWith = function (fn) {
```

Comment:
Basically i tried several "workarounds" trying to not perturb the default invocation. 
@satazor yes, when i tried a similar solution to your suggestion when, for example extending an default property onto the object with a decorator, the Proxy was "hijacking" (this is why i added the decorator exposure to the instance, to bypass a situation like this).